contracts/src/receipt.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright 2025 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use alloy_primitives::Bytes;
+use risc0_aggregation::{
+    decode_set_inclusion_seal, SetInclusionDecodingError, SetInclusionEncodingError,
+    SetInclusionReceipt,
+};
+use risc0_zkvm::{FakeReceipt, InnerReceipt, Receipt as Risc0Receipt, ReceiptClaim};
+use thiserror::Error;
+
+use crate::{
+    encode_seal,
+    groth16::decode_groth16_seal,
+    selector::{SelecorType, Selector, SelectorError},
+};
+
+/// Wrapper around different types of receipts.
+pub enum ReceiptType {
+    Risc0(Risc0Receipt),
+    SetInclusion(SetInclusionReceipt<ReceiptClaim>),
+}
+
+impl ReceiptType {
+    /// Encode the receipt as a seal.
+    pub fn encode_seal(&self) -> Result<Vec<u8>, SetInclusionEncodingError> {
+        match self {
+            ReceiptType::Risc0(receipt) => {
+                encode_seal(receipt).map_err(|_| SetInclusionEncodingError::UnsupportedReceiptType)
+            }
+            ReceiptType::SetInclusion(receipt) => receipt.abi_encode_seal(),
+        }
+    }
+
+    /// Get the receipt if it is a RISC0 receipt.
+    pub fn receipt(&self) -> Option<Risc0Receipt> {
+        match self {
+            ReceiptType::Risc0(receipt) => Some(receipt.clone()),
```

Comment:
Is there a reason this would not work? With this, they could clone it if needed or use it as a reference.
```suggestion
    pub fn receipt(&self) -> Option<&risc0_zkvm::Receipt> {
        match self {
            ReceiptType::Risc0(receipt) => Some(receipt),
```